test/extended/apiserver/latency.go

```diff
@@ -0,0 +1,79 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	promhelper "github.com/openshift/origin/test/extended/util/prometheus"
+)
+
+var _ = g.Describe("[sig-instrumentation][Late] API Server latency", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc               = exutil.NewCLIWithoutNamespace("apiserver-latency")
+		url, bearerToken string
+	)
+
+	g.BeforeEach(func() {
+		err := exutil.WaitForAnImageStream(
+			oc.AdminImageClient().ImageV1().ImageStreams("openshift"), "tools",
+			exutil.CheckImageStreamLatestTagPopulated, exutil.CheckImageStreamTagNotFound,
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		var ok bool
+		url, _, bearerToken, ok = promhelper.LocatePrometheus(oc)
+		if !ok {
+			e2e.Failf("Prometheus could not be located on this cluster, failing prometheus test")
+		}
+	})
+
+	g.It("should collect apiserver request latency metrics", func() {
+		ns := oc.SetupNamespace()
+
+		execPod := exutil.CreateExecPodOrFail(oc.AdminKubeClient(), ns, "execpod")
+		defer func() {
+			oc.AdminKubeClient().CoreV1().Pods(ns).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
+		}()
+
+		// consider series since the beginning of the test
+		testDuration := exutil.DurationSinceStartInSeconds().String()
+
+		resp, err := promhelper.RunQuery(
+			fmt.Sprintf(`
+				rate(apiserver_request_duration_seconds_sum{job="apiserver",verb=~"LIST|GET"}[%s])
```

Comment:
how about we separate the verbs: get, list, update, patch, delete should all be relatively constant time.